### PR TITLE
feat: expoJudgingSession/{id}/teams GET endpoint

### PR DESCRIPTION
**Pull request by Ansaar-Syed**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### CODE REVIEW

File: packages/api/src/api/expoJudgingSession/id/teams/get.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    if (!ejs) {
+      res.sendStatus(404);
+    }
+
+    await em.populate(judge, ['expoJudgingSessionContexts']);
+
+    const validSession = judge.expoJudgingSessionContexts
+      .getItems()
+      .filter((ejsc) => ejsc.expoJudgingSession.id === ejsId);
```
Comment (SpencerKaiser) [inline packages/api/src/api/expoJudgingSession/id/teams/get.ts:24]: Use the find method to get the one and only context matching your ID; doing this makes the below a lot simpler
```suggestion
      .find((ejsc) => ejsc.expoJudgingSession.id === ejsId);
```

File: packages/api/src/api/expoJudgingSession/id/teams/get.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    if (validSession.length) {
+      const currentTeam = validSession[0]?.currentProject;
+      const prevTeam = validSession[0]?.previousProject;
+
+      res.send({ currentTeam, prevTeam });
+    } else {
+      res.sendStatus(404);
```
Comment (SpencerKaiser) [inline packages/api/src/api/expoJudgingSession/id/teams/get.ts:32]: This should be a 403; the judge tried to make a vote against a session they aren't associated with
```suggestion
      res.sendStatus(403);
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]

File: packages/api/src/api/expoJudgingSession/id/projects/get.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { Request, Response } from 'express';
+import { ExpoJudgingSession } from '@hangar/database';
+import { logger } from '../../../../utils/logger';
+
+export const get = async (req: Request, res: Response) => {
+  // sone functionality
```
Comment (SpencerKaiser) [inline packages/api/src/api/expoJudgingSession/id/projects/get.ts:6]:
```suggestion
```

File: packages/api/src/api/expoJudgingSession/id/projects/get.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    if (validEjsContext) {
+      const { currentProject, previousProject } = validEjsContext;
+
+      res.send({ currentProject, previousProject });
+    } else {
+      res.sendStatus(403);
+    }
+  } catch (error) {
+    logger.error('Failed to query Expo Judging Session', error);
```
Comment (SpencerKaiser) [inline packages/api/src/api/expoJudgingSession/id/projects/get.ts:34]:
```suggestion
    logger.error('Failed to query Expo Judging Session Context', error);
```

File: packages/api/src/api/expoJudgingSession/id/projects/index.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { Router } from 'express';
+import { judgeMiddleware } from '../../../../middleware/judgeMiddleware';
+import { get } from './get';
+
+export const projects = Router();
+
+projects.get('', judgeMiddleware, get);
```
Comment (SpencerKaiser) [inline packages/api/src/api/expoJudgingSession/id/projects/index.ts:7]:
```suggestion
projects.get('', get);
```

File: packages/api/tests/api/expoJudgingSession/id/projects/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+jest.mock('../../../../../src/api/expoJudgingSession/id/projects/get', () => ({
+  get: createMockHandler(),
+}));
+
+jest.mock('../../../../../src/middleware/judgeMiddleware', () => ({
+  judgeMiddleware: createMockNext(),
+}));
+
+const judgeMiddlewareMock = getMock(judgeMiddleware);
```
Comment (SpencerKaiser) [inline packages/api/tests/api/expoJudgingSession/id/projects/index.test.ts:16]:
```suggestion
```

File: packages/api/tests/api/expoJudgingSession/id/projects/index.test.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+describe('id router', () => {
+  it('registers the get handler', async () => {
+    await jest.isolateModulesAsync(async () => {
+      const { projects } = await import('../../../../../src/api/expoJudgingSession/id/projects');
+      const app = express();
+      app.use(projects);
+
+      const res = await supertest(app).get('');
+      expect(res.status).toEqual(200);
+      expect(judgeMiddlewareMock).toBeCalledTimes(1);
```
Comment (SpencerKaiser) [inline packages/api/tests/api/expoJudgingSession/id/projects/index.test.ts:27]:
```suggestion
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]

File: packages/api/tests/api/expoJudgingSession/id/projects/index.test.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import supertest from 'supertest';
+import { getMock } from '../../../../testUtils/getMock';
+import { createMockHandler } from '../../../../testUtils/expressHelpers/createMockHandler';
+import { createMockNext } from '../../../../testUtils/expressHelpers/createMockNext';
+import { judgeMiddleware } from '../../../../../src/middleware/judgeMiddleware';
+
+jest.mock('../../../../../src/api/expoJudgingSession/id/projects/get', () => ({
+  get: createMockHandler(),
+}));
+
```
Comment (Ansaar-Syed) [inline packages/api/tests/api/expoJudgingSession/id/projects/index.test.ts:11]:
```suggestion
```

**Review by SpencerKaiser — APPROVED** [review]

File: packages/api/tests/api/expoJudgingSession/id/projects/index.test.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import express from 'express';
+import supertest from 'supertest';
+import { getMock } from '../../../../testUtils/getMock';
+import { createMockHandler } from '../../../../testUtils/expressHelpers/createMockHandler';
+import { createMockNext } from '../../../../testUtils/expressHelpers/createMockNext';
+import { judgeMiddleware } from '../../../../../src/middleware/judgeMiddleware';
```
Comment (SpencerKaiser) [inline packages/api/tests/api/expoJudgingSession/id/projects/index.test.ts:6]:
```suggestion
import { createMockHandler } from '../../../../testUtils/expressHelpers/createMockHandler';
import { createMockNext } from '../../../../testUtils/expressHelpers/createMockNext';
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]

**Review by SpencerKaiser — APPROVED** [review]

**Review by SpencerKaiser — APPROVED** [review]

**Pull request merged**